helm-chart-sources/pulsar/templates/function/function-configmap.yaml

```diff
@@ -114,7 +114,6 @@ data:
     authenticationEnabled: "true"
     authorizationEnabled: "true"
     authorizationProvider: "org.apache.pulsar.broker.authorization.PulsarAuthorizationProvider"
-    authenticationProviders: [org.apache.pulsar.broker.authentication.AuthenticationProviderToken, org.apache.pulsar.broker.authentication.AuthenticationProviderTls]
```

Comment:
I was having problems when trying to override this configuration. I removed this line and attempted to retain the functionality by adding the following to the `function-extra-configmap.yaml`:

```yaml
  authenticationProviders: "{{ .Values.function.authenticationProviders }}"
```

And I defined `.Values.function.authenticationProviders` in the `values.yaml` file. Since the function worker's pod config loads `function-extra-configmap.yaml` as environment variables and we run `bin/gen-yml-from-env.py conf/functions_worker.yml`, the extra configmap would supersede the original value in the config map you reference above. I thought this was working locally, but I could be wrong. Does this context help?

Reply:
@michaeljmarshall Pulsar has been a mess over the years in where it reads config, especially the function worker. The config is supposed to be read from `function_worker.yaml` but past bugs lead me to add it in both places to make sure it got picked up. I *think* `function_worker.yaml` is the right location now. And in my deployment, I keep getting 401 errors for sources and sink commands.

Reply:
@cdbartholomew - thanks for the background. I realize now that we can just add in the string from the `Values` file. I will get a change added this morning.